commons/src/main/java/org/infinispan/commons/api/functional/Param.java

```diff
@@ -44,6 +44,29 @@
    P get();
 
    /**
+    * Should this paramater go over the wire?
+    */
+   SerializeMethod serializeMethod();
```

Comment:
This is not how we deal with an objects serialization/marshalling characteristics. Instead, if a param instance needs to be serialized, an Externalizer should be provided for it (examples in the codebase) and then the Externalizer it's either caught by annotation processing (e.g. @SerializedWith) or it's preregistered. The externalizer itself can decide how it serialize it using whichever method it wants. A param implementation, if you want something lazy, can implement Serializable and that's it. So, there's no need to add anything to the Param interface, which has been designed with the bare minimum required to do its job.